cpp/src/gandiva/engine.cc

```diff
@@ -112,9 +113,20 @@ void Engine::InitOnce() {
   }
   ARROW_LOG(INFO) << "Detected CPU Name : " << cpu_name.str();
   ARROW_LOG(INFO) << "Detected CPU Features:" << cpu_attrs_str;
+
+  std::string engine_str = cpu_name.str();
+  engine_str += cpu_attrs_str;
+  engine_str_ = strdup(engine_str.c_str());
+
   llvm_init = true;
 }
 
+char* Engine::ToString() {
```

Comment:
also use a different name for this method..since we are only resulting the cpu details